File: fstcomp/examples/Modification/GPL/EdgeColoring/_mod/Edge.java
package GPL;

import java.util.LinkedList;

public class Edge extends Neighbor implements EdgeIfc {

    // TODO mod
    private int color;

    /**
     * @return the color
     */
    public int getColor() {
	return color;
    }

    /**
     * @param color
     *            the color to set
     */
    public void setColor(int color) {
	this.color = color;
    }

    // TODO /mod
    public Vertex start;

    public void EdgeConstructor(Vertex the_start, Vertex the_end) {
	start = the_start;
	end = the_end;
    }

    public void adjustAdorns__wrappee__UndirectedWithEdges(EdgeIfc the_edge) {
    }

    public void adjustAdorns(EdgeIfc the_edge) {
	setWeight(the_edge.getWeight());
	adjustAdorns__wrappee__UndirectedWithEdges(the_edge);
    }

    public void setWeight(int weight) {
	this.weight = weight;
    }

    public int getWeight() {
	return this.weight;
    }

    public Vertex getOtherVertex(Vertex vertex) {
	if (vertex == start) {
	    return end;
	} else if (vertex == end) {
	    return start;
	} else {
	    return null;
	}
    }

    public Vertex getStart() {
	return start;
    }

    public Vertex getEnd() {
	return end;
    }

    public void display__wrappee__UndirectedWithEdges() {
	System.out.println(" start=" + start.getName() + " end="
		+ end.getName());
    }

    public void display() {
	System.out.print(" Weight=" + weight);
	// TODO mod
	System.out.print(" color=" + color);
	// TODO /mod
	display__wrappee__UndirectedWithEdges();
    }

    private int weight;

    public void EdgeConstructor(Vertex the_start, Vertex the_end, int the_weight) {
	EdgeConstructor(the_start, the_end);
	weight = the_weight;
    }

}

File: fstcomp/examples/Modification/GPLCSharp/EdgeColoringTestOut/Edge.cs

using System.Collections; 
public class  Edge  : Neighbor, EdgeIfc {
	 public  Vertex start;  public  void EdgeConstructor( Vertex the_start, Vertex the_end ) { start = the_start; end = the_end; }  public  void adjustAdorns( EdgeIfc the_edge ) { }  public  void setWeight(int weight) { }  public  int GetWeight( ) { return 0; }  public  Vertex GetOtherVertex( Vertex vertex ) { if( vertex == start ) { return end; } else if(vertex == end) { return start; } else { return null; } }  public  Vertex GetStart( ) { return start; }  public  Vertex GetEnd( ) { return end; } private  int color; public  int getColor() {return color;} public  void setColor(int color) {this.color = color;}  public  void display__wrappee__DFS( ) { System.Console.Out.WriteLine( " start=" + start.GetName() + " end=" + end.GetName( ) ); } public  void display() {System.Console.Out.Write(" color=" + color);display__wrappee__DFS();}
}
